File: Java/tollcalculator/src/main/java/com/arngrimur/evolve/Vehicle.java
package com.arngrimur.evolve;

public enum Vehicle {
    MOTORBIKE("Motorbike"),
    TRACTOR("Tractor"),
    EMERGENCY("Emergency"),
    DIPLOMAT("Diplomat"),
    FOREIGN("Foreign"),
    MILITARY("Military"),
	CAR("Car");
    private final String type;

    Vehicle(String type) {
      this.type = type;
    }

    public String getType() {
      return type;
    }
    
    
    
  }

File: Go/vehicle.go
package main

import "EvolveTechnology/toll-calculator/utils"

var tollFreeVehicles = []string{"Motorbike", "Tractor", "Emergency", "Diplomat", "Foreign", "Military"}

type Vehicle interface {
	GetType() string
}

type (
	Motorbike        struct{}
	Car              struct{}
	Tractor          struct{}
	EmergencyVehicle struct{}
	DiplomatVehicle  struct{}
	ForeignVehicle   struct{}
	MilitaryVehicle  struct{}
)

func (m Motorbike) GetType() string         { return "Motorbike" }
func (c Car) GetType() string               { return "Car" }
func (t Tractor) GetType() string           { return "Tractor" }
func (ev EmergencyVehicle) GetType() string { return "Emergency" }
func (dv DiplomatVehicle) GetType() string  { return "Diplomat" }
func (fv ForeignVehicle) GetType() string   { return "Foreign" }
func (mv MilitaryVehicle) GetType() string  { return "Military" }

func IsTollFreeVehicle(v Vehicle) bool {
	return utils.Contains(tollFreeVehicles, v.GetType())
}
